napari/components/add_layers_mixin.py

```diff
@@ -66,6 +66,26 @@ def add_layer(self, layer: layers.Layer) -> layers.Layer:
             self.reset_view()
         return layer
 
+    def add_layergroup(
+        self, layergroup: layers.Layergroup
+    ) -> layers.Layergroup:
+        """Add a layergroup to the viewer.
+
+        Parameters
+        ----------
+        layergroup : :class:`napari.layers.Layergroup`
+            Layergroup to add.
+
+        Returns
+        -------
+        layergroup : :class:`napari.layers.Layergroup` or list
+            The layergroup that was added (same as input).
+        """
+        # FIXME
+        for layer in layergroup:
+            self.add_layer(layer)
+        return layergroup
+
```

Comment:
Is this needed? I'm hoping that we will be able to encapsulate everything in add_layer.

Reply:
There's a test checking for every Layer type, there is a corresponding `add_` method.
I can add an exception to that test in the case of layergroups, if you prefer.

Layergroups (as they stand now) break quite a few of these types of test.

Reply:
Failing tests are very much to be expected. But I guess there isn't one for Base, so however we prevent that, we should apply that to LayerGroup also.? I think?

Reply:
Nah, the check is for consistency among all **subclasses** of the base Layer class, so by definition that won't include Layer itself. I've added a conditional that excludes Layergroup from those tests for now (we can always change this later on).